contrib/cirrus/check_image.sh

```diff
@@ -36,4 +36,28 @@ do
         "$(systemctl list-unit-files --no-legend $REQ_UNIT)" = "$REQ_UNIT enabled" || let "RET+=1"
 done
 
+# Previously, perdiodic package updates were enabled and would trigger
+# soon after an Ubuntu VM booted, grabbing a specific lock file for
+# several minutes after setup_environment.sh ran (previous script here also).
+DPKGLCK=/var/lib/dpkg/lock
+if [[ "$OS_RELEASE_ID" == "ubuntu" ]] && [[ -r "$DPKGLCK" ]]
+then
+    item_test "The lock file $DPKGLCK exists" -r "$DPKGLCK" || let "RET+=1"
+    flock --exclusive --nonblock --no-fork $DPKGLCK sleep 1h &
+    sleep 1s  # just in case
+    FLOCKPID="$!"
+    item_test "Uncontested dpkg lock acquired" "$(jobs -r | wc -l)" -eq "1" || let "RET+=1"
```

Comment:
This seems racy. What would you think of testing (manually, in a 5-second loop perhaps, not with `item_test`) that `$(</proc/$FLOCKPID/cmdline) =~ ^sleep` ?

Reply:
Also maybe why I put the `sleep 1s`, but yeah, it does a fork/exec so cmdline will always have "flock...sleep" in it :disappointed: 

I can't think of an easy way to confirm the lock is held by flock and sleep.  I think the sub-shell flock-and-redirect trick (from flock man page) is needed here.

Reply:
From flock(1):
```
-F, --no-fork
      Do not fork before executing command.  Upon execution the  flock
      process is replaced by command which continues to hold the lock
```
IOW, after the lock is grabbed, flock exec()s the given command. On failure, of course, it exits. Therefore if `/proc/$FLOCKPID/cmdline` exists and begins with `sleep`, it is a guarantee that the lock succeeded.

Reply:
code deleted